moveit_core/src/moveit_core/__init__.py

```diff
@@ -0,0 +1,2 @@
+from pymoveit_core import *
+
```

Comment:
You shouldn't include the library here, but the local modules:
```suggestion
from . import collision_detection, kinematic_constraints, planning_scene, robot_model, robot_state, transforms
```
Otherwise, local extensions (in python code) don't become available. You don't yet have them, but for the future...

Reply:
I'm not sure, I think the line you suggest adding does nothing? And in my local testing we need the line `from pymoveit_core import *`, because otherwise we don't get any of the symbols to `pymoveit_core`.  For example try running `from moveit_core import load_robot_model`. As is this works, but with what you propose it does not. If you have examples that do/don't work that might help me understand.

I also tested adding `foo.py` to `moveit_core/src/moveit_core` with a function `test`, I can write `from moveit_core import foo` and `foo.test()` works fine. I also tested this in the submodules.

Reply:
For the examples you gave, my suggested line is not important, but for this it is:
```python
import pymoveit_core as core
core.robot_model.RobotModel()  # without, you couldn't access the contents of core here
```
The symbols from the (library) submodules are loaded via `<submodule>/__init__.py`:
```python
from pymoveit_core.<submodule> import *
# without the proposed change, you wouldn't be able to augment / override methods in a submodule like this:
def added_to_submodule():
    pass
```
Adding a new submodule `foo`, of course, works in any case.

Reply:
My thought was no one need reference pymoveit_core directly. you can just write moveit_core and get both the C++ bindings and any python code we add. This works, and I think is the same as what you wrote

```
import moveit_core
moveit_core.robot_model.RobotModel
```

I'm not sure what you mean by override. Like this? this works:

```
from moveit_core.kinematic_constraints import *                                                                                                                 

 constructGoalConstraints # original method

def constructGoalConstraints():
    print("you're mine now")
```

Reply:
I need to test again. I'm currently working on MTC python bindings and there, I needed the suggested changes...
I will postpone that for tomorrow.

Reply:
I tested and now I remember the exact open issue: With your approach you will not be able to access augmented python code like this:
```python
import moveit_core as core
core.robot_model.added_to_submodule()
```
Sorry for pointing initially into the wrong direction. As I prepared the fixup for testing anyway, you can find a corresponding PR [here](https://github.com/UM-ARM-Lab/moveit/pull/1).

Reply:
I've updated it to use `moveit.core`, the install works, and it is compatible with other packages, so when we add bindings for moveit_ros_planning for example, it should become something like `from moveit.planning import robot_model_loader`